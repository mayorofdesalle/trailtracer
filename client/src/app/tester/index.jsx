import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
