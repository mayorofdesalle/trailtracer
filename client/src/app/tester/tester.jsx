import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };