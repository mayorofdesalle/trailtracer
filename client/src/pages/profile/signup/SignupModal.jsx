import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import styled, { useTheme } from 'styled-components';

import { glassMorph } from '@components/mixins';
import { Button, Container, Icon, Text } from '@components/ui';
import { BentoBox, BentoGrid } from '@components/ui/bento';

import AuthButtons from '@pages/profile/components/AuthButtons';
import Separator from '@pages/profile/components/Separator';
import SignupForm from './SignupForm';

const SignupBentoGrid = styled(BentoGrid)`
    position: absolute;
	--margin: 1rem;
	top: var(--margin);
	left: var(--margin);
	height: calc(100% - 2 * var(--margin));
	width: calc(100% - 2 * var(--margin));
    ${glassMorph}
    border: none;
    z-index: 9999;

    grid-template-columns: repeat(3, 1fr);
    grid-template-rows: 1fr 10fr 1fr;
    grid-template-areas:
        '. .           .'
        '. SignupModal .'
        '. .           .';

    /* Medium screens and downwards */
    @media (max-width: ${({ theme }) => theme.breakpoints.medium}px) or (max-aspect-ratio: 1) {
        grid-template-columns: auto 1fr auto;
        grid-template-rows: auto 1fr auto;
    }
`;

const SignupModalInner = styled(BentoBox)`
    grid-area: SignupModal;
    position: absolute;
    width: 100%;
    height: 100%;
    background-color: ${({ theme }) => theme.colors.background};
    outline: 2px solid ${({ theme }) => theme.colors.primaryTransparent};
    flex-direction: column;
    justify-content: space-around;

    & > * {
        width: 80%;
    }
    
    & > ${Button} {
        position: absolute;
        right: clamp(1rem, min(2dvw, 2dvh), 4rem);
        top: clamp(1rem, min(2dvw, 2dvh), 4rem);
        padding: 0;
        --size: clamp(1.5rem, min(3dvw, 3dvh), 6rem);
        width: var(--size);
        height: var(--size);
    }
`;

const Wrapper = styled(Container)`
    height: fit-content;
    flex-direction: column;
`;

const SignupModal = ({ close, ...props }) => {
    const theme = useTheme();
    const { t } = useTranslation();

    return (
        <SignupBentoGrid>
            <SignupModalInner role='dialog'>
                <Button onClick={close}>
                    <Icon name="close-large" />
                </Button>
                <Wrapper>
                    <Text $color={theme.colors.primary} $heading $ratio={1.5}>{t('signupModal.title')}</Text>
                    <Text $color={theme.colors.text}>{t('signupModal.subtitle')}</Text>
                </Wrapper>
                <SignupForm />
                <Separator color={theme.colors.text} />
                <AuthButtons />
            </SignupModalInner>
        </SignupBentoGrid>
    );
};

SignupModal.propTypes = {
    close: PropTypes.func
};

export default SignupModal;