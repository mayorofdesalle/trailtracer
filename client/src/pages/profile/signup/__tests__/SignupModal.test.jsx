import { describe, expect, it, render, screen } from '@app/tester';

import SignupModal from '../SignupModal';

describe('SignupModal', () => {

    it('should render correctly', () => {
        render(<SignupModal />);
        const signupModal = screen.getByRole('dialog');
        expect(signupModal).toBeInTheDocument();
        expect(signupModal).toHaveStyle({ gridArea: 'SignupModal' });

        const form = screen.getByRole('form');
        expect(form).toBeInTheDocument();

        const buttons = screen.getAllByRole('button');
        expect(buttons).toHaveLength(5);
        buttons.forEach(button => expect(button).toBeInTheDocument());
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<SignupModal />);
        expect(asFragment()).toMatchSnapshot();
    });

});