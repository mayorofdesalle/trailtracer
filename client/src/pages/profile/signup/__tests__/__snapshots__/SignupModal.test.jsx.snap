// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignupModal > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-kFCroH sc-blmEgr kHwwsn cZKACD IbSiZ"
    role="grid"
  >
    <div
      class="sc-guDLey sc-dmyCSP sc-ifyrAs kHwwsn fkKYEp uzUSZ"
      role="dialog"
    >
      <button
        class="sc-beySPh zbkUW"
        type="button"
      >
        <svg
          class="sc-eDLKkx kBnHHP"
          fill="currentColor"
          role="img"
        >
          <title>
            close-large
          </title>
          <use
            href="http://localhost:3000/src/assets/icons.svg#close-large"
          />
        </svg>
      </button>
      <div
        class="sc-guDLey sc-dJGMql kHwwsn kFqbDj"
      >
        <div
          class="sc-gLLuof hnPDDY"
        >
          signupModal.title
        </div>
        <div
          class="sc-gLLuof fyoZns"
        >
          signupModal.subtitle
        </div>
      </div>
      <form
        class="sc-fmKFGs iMNvzB"
        role="form"
      >
        <div
          class="sc-guDLey sc-fHejqy kHwwsn jvnvBG"
        >
          <div
            class="sc-gLLuof jmevau"
          >
            forms.email
          </div>
          <div
            class="sc-guDLey sc-ktwOfi kHwwsn ecTIsN"
          >
            <label
              for="email"
              hidden=""
            >
              forms.email
            </label>
            <input
              autocomplete="email"
              class="sc-brSamD fskSZa"
              data-tooltip-id="email"
              id="email"
              name="email"
              placeholder="forms.email"
              type="email"
            />
            <svg
              class="sc-eDLKkx kBnHHP"
              fill="#435049"
              role="img"
            >
              <title>
                at-fill
              </title>
              <use
                href="http://localhost:3000/src/assets/icons.svg#at-fill"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-guDLey sc-fHejqy kHwwsn jvnvBG"
        >
          <div
            class="sc-gLLuof jmevau"
          >
            forms.new-password
          </div>
          <div
            class="sc-guDLey sc-ktwOfi kHwwsn ecTIsN"
          >
            <label
              for="password"
              hidden=""
            >
              forms.new-password
            </label>
            <input
              autocomplete="new-password"
              class="sc-brSamD iAHoYb"
              data-tooltip-id="password"
              id="password"
              name="new-password"
              placeholder="forms.new-password"
              type="password"
            />
            <svg
              class="sc-eDLKkx kBnHHP"
              fill="#435049"
              role="img"
            >
              <title>
                key-fill
              </title>
              <use
                href="http://localhost:3000/src/assets/icons.svg#key-fill"
              />
            </svg>
          </div>
        </div>
        <button
          class="sc-beySPh zbkUW"
          type="submit"
        >
          <div
            class="sc-gLLuof ldUHqL"
          >
            forms.continue
          </div>
        </button>
      </form>
      <div
        class="sc-guDLey sc-eTNRI kHwwsn jVeJPO"
        role="separator"
      >
        <hr
          class="sc-csKJxZ hOxJoW"
          role="none"
        />
        <div
          class="sc-gLLuof fyoZns"
        >
          signinPage.auths
        </div>
        <hr
          class="sc-csKJxZ hOxJoW"
          role="none"
        />
      </div>
      <div
        class="sc-guDLey sc-irLvIq kHwwsn bdPjGM"
      >
        <button
          class="sc-beySPh zbkUW"
          type="button"
        >
          <svg
            class="sc-eDLKkx kBnHHP"
            fill="currentColor"
            role="img"
          >
            <title>
              google-fill
            </title>
            <use
              href="http://localhost:3000/src/assets/icons.svg#google-fill"
            />
          </svg>
        </button>
        <button
          class="sc-beySPh zbkUW"
          type="button"
        >
          <svg
            class="sc-eDLKkx kBnHHP"
            fill="currentColor"
            role="img"
          >
            <title>
              facebook-fill
            </title>
            <use
              href="http://localhost:3000/src/assets/icons.svg#facebook-fill"
            />
          </svg>
        </button>
        <button
          class="sc-beySPh zbkUW"
          type="button"
        >
          <svg
            class="sc-eDLKkx kBnHHP"
            fill="currentColor"
            role="img"
          >
            <title>
              twitter-fill
            </title>
            <use
              href="http://localhost:3000/src/assets/icons.svg#twitter-fill"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
